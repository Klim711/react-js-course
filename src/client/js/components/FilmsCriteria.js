// @flow
import React from 'react';

type FilmsCriteriaProps = {
  criteriaValue: string;
  criteria: string;
}

const FilmsCriteria = (props: FilmsCriteriaProps) => (
  <div className="films-header">
    Films by {props.criteriaValue} {props.criteria}
  </div>
);

export default FilmsCriteria;
