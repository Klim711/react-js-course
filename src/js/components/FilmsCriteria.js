import React, {Component} from 'react';

const FilmsCriteria = function(props) {
  return (
    <div className="films-header">
      Films by {props.criteriaValue} {props.criteria}
    </div>
  );
}

export default FilmsCriteria;