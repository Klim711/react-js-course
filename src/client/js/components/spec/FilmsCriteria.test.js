import React from 'react';
import renderer from 'react-test-renderer';

import FilmsCriteria from '../FilmsCriteria';

describe('FilmsCriteria component', () => {
  it('matches to snapshot', () => {
    const renderedValue = renderer.create(<FilmsCriteria criteriaValue="mockValue" criteria="mockCriteria"/>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
