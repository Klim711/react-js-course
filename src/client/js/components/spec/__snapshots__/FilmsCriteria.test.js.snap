// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsCriteria component matches to snapshot 1`] = `
<div
  className="films-header"
>
  Films by 
  mockValue
   
  mockCriteria
</div>
`;
